enterprise/cli/dbcrypt_rotate.go

```diff
@@ -0,0 +1,152 @@
+//go:build !slim
+
+package cli
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/base64"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/cli"
+	"github.com/coder/coder/v2/cli/clibase"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/enterprise/dbcrypt"
+
+	"golang.org/x/xerrors"
+)
+
+func (*RootCmd) dbcryptRotate() *clibase.Cmd {
+	var (
+		vals = new(codersdk.DeploymentValues)
+		opts = vals.Options()
+	)
+	cmd := &clibase.Cmd{
+		Use:   "dbcrypt-rotate --postgres-url <postgres_url> --external-token-encryption-keys <new-key>,<old-keys>",
+		Short: "Rotate database encryption keys",
+		Options: clibase.OptionSet{
+			*opts.ByName("Postgres Connection URL"),
+			*opts.ByName("External Token Encryption Keys"),
+		},
+		Middleware: clibase.Chain(
+			clibase.RequireNArgs(0),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			ctx, cancel := context.WithCancel(inv.Context())
+			defer cancel()
+			logger := slog.Make(sloghuman.Sink(inv.Stdout))
+			if ok, _ := inv.ParsedFlags().GetBool("verbose"); ok {
+				logger = logger.Leveled(slog.LevelDebug)
+			}
+
+			if vals.PostgresURL == "" {
+				return xerrors.Errorf("no database configured")
+			}
+
+			switch len(vals.ExternalTokenEncryptionKeys) {
+			case 0:
+				return xerrors.Errorf("no external token encryption keys provided")
+			case 1:
+				logger.Info(ctx, "only one key provided, data will be re-encrypted with the same key")
```

Comment:
Why support this? Is this so people can upgrade from no encryption to encryption? If so the log message is confusing

Reply:
Correct - the primary reason for this is to allow immediately encrypting data.
I'll amend the log message.